html/infrastructure/common-dom-interfaces/collections/htmlallcollection.html

```diff
@@ -181,9 +207,36 @@
 }, "legacy caller with invalid \"array index property name\"");
 
 test(function() {
+  assert_equals(document.all(undefined), divs[3]);
+}, "legacy caller with undefined");
+
+test(function() {
+  assert_equals(document.all(null), divs[4]);
+}, "legacy caller with null");
+
+test(function() {
   assert_equals(document.all(), null);
 }, "legacy caller with no argument");
 
+test(function() {
+  if (typeof Proxy !== "undefined") {
+    assert_throws(new TypeError(), function() {
+      new (new Proxy(document.all, {
+        construct: function() {
+          return {};
+        }
+      }));
+    });
+  }
+  assert_throws(new TypeError(), function() {
+    new document.all("picture");
+  });
+}, "legacy caller is not a constructor");
+
+test(function() {
+  assert_equals(Function.prototype.call.call(document.all, {}, "043"), spans[1]);
+}, "legacy caller with arbitrary this value");
```

Comment:
Can you add one where the this value is a platform object? E.g., document.body?

Reply:
Sure.